Provis.Core/Services/UserService.cs

```diff
@@ -41,5 +38,56 @@ public async Task<UserPersonalInfoDTO> GetUserPersonalInfoAsync(string userId)
 
             return userPersonalInfo;
         }
+        public async Task ChangeNameAsync(string userId, UserChangeNameDTO userChangeNameDTO)
+        {
+            var user = await _userRepository.GetByKeyAsync(userId);
+
+            if (user == null)
+            {
+                throw new HttpException(System.Net.HttpStatusCode.NotFound, "User with Id not exist");
+            }
+
+            user.Name = userChangeNameDTO.Name;
+
+            await _userRepository.UpdateAsync(user);
+
+            await _userRepository.SaveChangesAsync();
+
+            await Task.CompletedTask;
+        }
+        public async Task ChangeSurnameAsync(string userId, UserChangeSurnameDTO userChangeSurnameDTO)
+        {
+            var user = await _userRepository.GetByKeyAsync(userId);
+
+            if (user == null)
+            {
+                throw new HttpException(System.Net.HttpStatusCode.NotFound, "User with Id not exist");
```

Comment:
Are we sure that core service should know anything about WEB ?

Reply:
we will deal with this in the following sprints

Reply:
we will deal with this in the following sprints in a separate task